ballerina/http_connection.bal

```diff
@@ -139,6 +140,7 @@ public isolated client class Caller {
         } else if message is error {
             if message is ApplicationResponseError {
                 InternalServerError err = {
+                    mediaType: returnMediaType is ()? mime:APPLICATION_JSON: returnMediaType,
```

Comment:
@chamil321 are you okay with this change ?

Reply:
We pass the `returnMediaType` to the `createStatusCodeResponse()` method and set the content type according to below order. So if we override it, probably `mediaType` field will get high priority when user set it or not(I mean when () is passed.)
```
    // Update content-type header according to the priority. (Highest to lowest)
    // 1. MediaType field in response record
    // 2. Payload annotation mediaType value
    // 3. The content type header included in headers field
    // 4. Default content type related to payload
```

Reply:
Since we are setting the header to Error response, I think it's fine.

Reply:
Correction on the above comment. 
I think this change is wrong. With this change all every data type gets only the `mime:APPLICATION_JSON` as it's content type whenever the return type is not mentioned